### ISSUE TITLE
Able to "delete" pmfolders that are classified as "Non-removable"

### ISSUE THREAD

**Issue by Jockerino** [mybb/mybb#3729]
Hello,

After some testing on the pm system I managed to find an issue with pmfolders. As you may see in this screenshot https://i.imgur.com/qaVtg01.png it clearly says that those folders cannot be removed. However, after messing around with the POST request that is sent while "Editing Folders" I managed to get those removed. 

`POST /mybb/private.php HTTP/1.1
Host: localhost.
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:67.0) Gecko/20100101 Firefox/67.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Referer: http://localhost./mybb/private.php?action=folders
Content-Type: application/x-www-form-urlencoded
Content-Length: 285
Connection: close
Cookie: mybb[lastvisit]=1560776956; mybb[lastactive]=1560777029; sid=e8814aa75ab94641608900392433da90; loginattempts=1; mybbuser=1_zsvdx6hy7aNN05NY0Nkg5y4fYoYvrlYiaGKJicYMRGtSFBzxS4
Upgrade-Insecure-Requests: 1

my_post_key=575bcb39715fccb640f0e2878392ed25&folder%5B0%5D=Inbox&folder%5B1%5D=Unread&folder%5B2%5D=Sent+Items&folder%5B3%5D=Drafts&folder%5B4%5D=Trash+Can&folder%5Bnew1%5D=&folder%5Bnew2%5D=&folder%5Bnew3%5D=&folder%5Bnew4%5D=&folder%5Bnew5%5D=&action=do_folders&submit=Update+Folders`

That's the original POST request that will work without any issues.

If you edit out the pmfolders and change the numbers of the folders then the folders will be removed as shown here.

`POST /mybb/private.php HTTP/1.1
Host: localhost.
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:67.0) Gecko/20100101 Firefox/67.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Referer: http://localhost./mybb/private.php?action=folders
Content-Type: application/x-www-form-urlencoded
Content-Length: 194
Connection: close
Cookie: mybb[lastvisit]=1560776956; mybb[lastactive]=1560777029; sid=e8814aa75ab94641608900392433da90; loginattempts=1; mybbuser=1_zsvdx6hy7aNN05NY0Nkg5y4fYoYvrlYiaGKJicYMRGtSFBzxS4
Upgrade-Insecure-Requests: 1

my_post_key=575bcb39715fccb640f0e2878392ed25&folder%5B0%5D=test&folder%5Bnew1%5D=&folder%5Bnew2%5D=&folder%5Bnew3%5D=&folder%5Bnew4%5D=&folder%5Bnew5%5D=&action=do_folders&submit=Update+Folders`

What I did was remove the rest folders from the POST request and the result was this https://i.imgur.com/Nb6kaSs.png

**Comment by effone** [comment]
You were able to manipulate a post request and bypassed whitespace check in naming anong with allowed key > 4 only!
Interesting ...

**Comment by Jockerino** [comment]
Indeed, not really an important thing but would be nice if this was fixed.

**Comment by Sama34** [comment]
I will reopen this because it is legit: people can remove restricted folders by manipulating the form data.
